http-client/Network/HTTP/Client/Types.hs

```diff
@@ -263,26 +269,51 @@ data Cookie = Cookie
 newtype CookieJar = CJ { expose :: [Cookie] }
   deriving (Read, Show, T.Typeable)
 
--- This corresponds to step 11 of the algorithm described in Section 5.3 \"Storage Model\"
-instance Eq Cookie where
-  (==) a b = name_matches && domain_matches && path_matches
-    where name_matches = cookie_name a == cookie_name b
-          domain_matches = CI.foldCase (cookie_domain a) == CI.foldCase (cookie_domain b)
-          path_matches = cookie_path a == cookie_path b
-
-instance Ord Cookie where
-  compare c1 c2
+-- | Instead of '(==)'.
```

Comment:
On all Haddocks, please add a `@since`. To go along with that, please include a major version bump and changelog entry.

Reply:
I'm guessing `0.6.5`?  e35735c7b6f33539b5262d642c06662c98f6027e

Reply:
No, that would be a minor version bump. It should be 0.7.0.

Reply:
Thanks, but this is still missing a bump in the cabal file and changelog entry. Also, I haven't looked into it, but all of the CI jobs are failing right now.

Reply:
Sorry, i don't know your process all that well, please bear with me.  :)

Tests passed locally, i don't think i'm doing anything ghc-version-sensitive, and the CI errors look unrelated to me:

```
unix-compat         > cbits/HsUnixCompat.c: In function ‘unix_major’:
unix-compat         > 
unix-compat         > /tmp/stack-66af33c894a42cac/unix-compat-0.5.0.1/cbits/HsUnixCompat.c:9:13: error:
unix-compat         >      warning: In the GNU C Library, "major" is defined
unix-compat         >      by <sys/sysmacros.h>. For historical compatibility, it is
unix-compat         >      currently defined by <sys/types.h> as well, but we plan to
unix-compat         >      remove this soon. To use "major", include <sys/sysmacros.h>
unix-compat         >      directly. If you did not intend to use a system-defined macro
unix-compat         >      "major", you should undefine it after including <sys/types.h>.
unix-compat         >          return major(dev);
unix-compat         >                  ^~~~~~~~~~                                                                                                                                                                                                                                                                                                                                                              
unix-compat         >   |
unix-compat         > 9 |     return major(dev);
unix-compat         >   |             ^
```

If you want me to I will try to have a look, but that may take more time than this mini-PR so far.  Or maybe you have a hint?

Reply:
Sorry, forgot to run *all* tests in the repo.  I'll take another look.

Reply:
> Sorry, i don't know your process all that well, please bear with me. :)

No worries at all, thank you for your quick responses on all my requests!